updater/lib/dependabot/api_client.rb

```diff
@@ -170,7 +170,7 @@ def mark_job_as_processed(base_commit_sha)
       span&.finish
     end
 
-    sig { params(dependencies: T::Array[T::Hash[Symbol, T.untyped]], dependency_files: T::Array[DependencyFile]).void }
+    sig { params(dependencies: T::Array[T::Hash[Symbol, T.untyped]], dependency_files: T::Array[String]).void }
```

Comment:
I guess runtime validation isn't working for this part of the code because it's been wrong and I don't see any TypeErrors for it. It would happen for each job so we'd definitely notice.

Reply:
How come we're exposing this now, and can we verify that it's actually always taking a string?

Reply:
You can see in all the smoke tests, the data for the `update_dependency_list` endpoint's `dependency_files` is always an array of strings: https://github.com/dependabot/smoke-tests/blob/258269368c3c68116cf569bfed8bbafa5fcdc8e5/tests/smoke-bundler.yaml#L97

> How come we're exposing this now

Now that we've got more and more files typed, things are falling into place and mistakes made earlier are getting fixed. If there was no runtime error then no problem I guess 🤷